src/cascadia/TerminalControl/SearchBoxControl.idl

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.TerminalControl
+{
+    [default_interface] runtimeclass SearchBoxControl : Windows.UI.Xaml.Controls.UserControl
+    {
+        SearchBoxControl();
+        Boolean GetGoForward();
+        Boolean GetIsCaseSensitive();
+        void SetFocusOnTextbox();
+        Boolean ContainsFocus();
+
+        event Windows.Foundation.TypedEventHandler<SearchBoxControl, String> Search;

```

Comment:
I wonder if we should just use a delegate instead.


```
delegate void SearchHandler(String query, Boolean forward, Boolean caseSensitive);
...
runtimeclass xxx {
    event SearchHandler Search();
}
```

This means we do not have to create an object, _and_ we don't need to use a TypedEventHandler.

I believe this is the "correct"(?) use of delegates?

Reply:
Then, the consumer can still add a search handler like this:

```
searchBoxControl.Search([](hstring query, bool forward, bool caseSensitive) {
})
```

Reply:
I can try this but it will take some time if we are not familiar with it ? I wonder if we can try this after checking-in the first working version ,as discussed before ?

Reply:
I am okay with checking this in and fixing it immediately afterwards. @zadjii-msft?

Reply:
(I do not believe that this is "broken", and getting user eyes on the experience would be more valuable than holding up over a few boolean properties.)

Reply:
Yea, that's fine with me. We should make sure a follow up task is filed.

Reply:
Created a separate issue for this: https://github.com/microsoft/terminal/issues/3942